src/edu/ucsb/nceas/metacat/dataone/MNodeService.java

```diff
@@ -3493,26 +3489,64 @@ public void publishIdentifier(Session session, Identifier identifier)
     }
 
     /**
-     * The admin API call to reindex partial or all documents in the instance
+     * The admin API call to reindex a list of documents in the instance.
      * @param session  the identity of requester. It must have administrative permissions
-     * @param identifiers  the list of objects' identifier which will be reindexed. If the parameter
-     *                     all is true, this parameter will be ignored.
-     * @param all  indicator if we need to reindex all objects in this instance
-     * @return true if the reindex request is scheduled; otherwise false
+     * @param identifiers  the list of objects' identifier which will be reindexed.
+     * @return true if the reindex request is scheduled. If something went wrong, an exception will
+     * thrown.
      * @throws ServiceFailure
      * @throws NotAuthorized
      * @throws NotImplemented
      * @throws InvalidRequest
      */
-    public boolean reindex(Session session, List<Identifier> identifiers, boolean all)
+    public Boolean reindex(Session session, List<Identifier> identifiers)
                              throws ServiceFailure, NotAuthorized, NotImplemented, InvalidRequest {
-        boolean scheduled = true;
+        Boolean scheduled = Boolean.valueOf(true);
+        String serviceFailureCode = "5901";
+        String notAuthorizedCode = "5902";
+        String notAuthorizedError ="The provided identity does not have permission to reindex "
+                                    + "objects on the Node: ";
+        checkAdminPrivilege(session, serviceFailureCode, notAuthorizedCode, notAuthorizedError);
+        handleReindexAction(identifiers);
+        return scheduled;
+    }
+
+    /**
+     * The admin API call to reindex all documents in the instance.
+     * @param session  the identity of requester. It must have administrative permissions
+     * @return true if the reindex request is scheduled. If something went wrong, an exception will
+     * thrown.
+     * @throws ServiceFailure
+     * @throws NotAuthorized
+     * @throws NotImplemented
+     * @throws InvalidRequest
+     */
+    public Boolean reindexAll(Session session) throws ServiceFailure, NotAuthorized,
+                                                                NotImplemented, InvalidRequest {
```

Comment:
Not all DataONE API providers implement optional methods like this, so that exception is there as a standard way to declare when a method is not supported. So it should stay in the declaration, even though Metacat implements the method.